zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/appliers/EventAppliers.java

```diff
@@ -182,11 +182,16 @@ private void registerDeploymentAppliers(final MutableProcessingState state) {
   }
 
   private void registerVariableEventAppliers(final MutableProcessingState state) {
-    final VariableApplier variableApplier = new VariableApplier(state.getVariableState());
+    final var variableState = state.getVariableState();
+    final var variableApplier = new VariableApplier(variableState);
     register(VariableIntent.CREATED, variableApplier);
     register(VariableIntent.UPDATED, variableApplier);
     register(VariableIntent.MIGRATED, new VariableMigratedApplier());
-    register(VariableDocumentIntent.UPDATED, NOOP_EVENT_APPLIER);
+    register(VariableDocumentIntent.UPDATING, new VariableDocumentUpdatingApplier(variableState));
+    register(VariableDocumentIntent.UPDATED, new VariableDocumentUpdatedApplier(variableState));
```

Comment:
💭 This changes the replay of this event, but as the implementation is a no-op for existing VariableDocument UPDATED events it's fine